src/grad.jl

```diff
@@ -8,33 +8,40 @@ Approximate the gradient of `f` at `xs...` using `fdm`. Assumes that `f(xs...)`
 """
 function grad end
 
-function grad(fdm, f, x::AbstractArray{T}) where T <: Number
+function _grad(fdm, f, x::AbstractArray{T}) where T <: Number
```

Comment:
Doesn't technically matter, but maybe change / remove the type constraint on `x`, since we know it's going to be an `Array` by virtue of the fact it's called from `grad`

Reply:
Could be a a few things actually.
But yes, the constraints are not needed

Reply:
I am going to keep it because the `T` is useful